.github/workflows/generate-tag.yml

```diff
@@ -0,0 +1,77 @@
+name: Verify repository and generate tags
+
+on: [pull_request_target]
+
+jobs: 
+  verify-repository-content:
+    name: Verify repository content
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - name: Generate repository content from version
+        run: |
+          bash bin/update-dockerfiles.sh
+          git diff -R | tee ${GITHUB_WORKSPACE}/pr.diff
+      - name: Generate tags for pr
+        run: |
+          python3 bin/tag-generator.py | tee ${GITHUB_WORKSPACE}/.tags
+      - name: Add Comment to PR
+        uses: actions/github-script@v5
+        with: 
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            commentText = 'Diff for ' + context.payload.pull_request.head.sha + ':\n\n'
+            needNewComment = true;
```

Comment:
Can you add some comments on what this is doing in the different blocks?

Reply:
Added. Some of the descriptions are available in `console.log`'s as well.